A Vagrant configuration for NCSA's nebula OpenStack cloud

jmatt opened the pull request:
A Vagrant configuration for NCSA's nebula OpenStack cloud + lsstsw.
- Uses @jhoblitt's lsstsw snapshots.
- Supports el6 and el7 machines.
- Improved README.md documentation.
- Added references to other resources.

If anyone else has ideas on how to improve it, please contribute them.

jmatt committed:
MVP that works with nebula.

	new file:   .gitignore
	new file:   README.md
	new file:   Vagrantfile

jmatt committed:
Improve the README.md.

	modified:   README.md

jmatt committed:
MVP to use vagrant + openstack + lsstsw.

    * Use @jhoblitt's nebula snapshot.
    * Support el6 and el7.
    * Update README.md by adding detail and removing
      `--provider=openstack` which isn't required when there is
      only one provider.

	modified:   README.md
	modified:   Vagrantfile

jmatt committed:
MVP to use vagrant + openstack + lsstsw.

    * Use @jhoblitt's nebula snapshot.
    * Support el6 and el7.
    * Update README.md by adding detail and removing
      `--provider=openstack` which isn't required when there is
      only one provider.

	modified:   README.md
	modified:   Vagrantfile

jmatt committed:
Improve the README.md.

	* Add Getting Credentials section.
	* Add More Resources section.

	modified:   README.md

jmatt committed:
Use UUIDs instead of the actual name to avoid ambiguity.

	modified:   Vagrantfile

jonathansick commented:
Hey @jmatt this is cool. :sunglasses: 

Could you add some instructions on where to go to install Vagrant? Near the top, something like:

> A vagrant configuration backed by NCSA's nebula OpenStack cloud.
> 
> Vagrant is a tool for running and managing virtual machines locally, or in this case, in clouds like Nebula.
> 
> You can download and install Vagrant by visiting http://www.vagrantup.com/downloads

Also in the readme you might want to standardize on the case of {N,n}ebula and {V,v}agrant when using it as a noun (not a command or object).

Also, maybe change the title "Getting Credentials" to "Getting Nebula Credentials"

I'll continue my review once I figure out how to get a Nebula account. 😑

jmatt committed:
Improve documentation in the README.md.

	* Add Nebula to the Getting Credentials section.
	* Add details about Vagrant.
	* Use the appropriate Vagrant/vagrant/nebula/Nebula.
	* Add more URLs in the More Resources section.

	modified:   README.md

jonathansick commented:
I recommend these changes to the "Getting Nebula Credentials" section (keep the bash part):

> You must have the appropriate credentials if you want to use Vagrant to make queries against the nebula OpenStack cloud. By far, the easiest way to obtain authentication credentials is to use the [Nebula Horizon dashboard](https://nebula.ncsa.illinois.edu/dashboard/project/access_and_security/):
> 1. Open the [Access and Security page](https://nebula.ncsa.illinois.edu/dashboard/project/access_and_security/) of the Nebula dashboard.
> 2. Click the **API Access** tab
> 3. Click on the **Download OpenStack RC File** button.
> 
> This will generate a file that you can `source` in your shell to populate the environment variables that Vagrant require to know where nebula's service endpoints and authentication information are. In the case of the LSST project, the file will be called `LSST-openrc.sh`.

jonathansick commented:
When you say "clone this repo", you might want to just put in the actual copy-and-pastable clone command

jmatt committed:
Improve the documentation in response to @jonathansick's recommendations.

	* Add Getting Vagrant section.
	* Reword Getting Nebula Credentials section.
	* Rename Using Vagrant section to be more specific. Create
	  your Vagrant machine.
	* Reference the LSST Software Stack directly.
	* Add Destroy your Vagrant machine section.
	* Add Available Vagrant Machines section.
	* Add Internals section.

	modified:   README.md

jmatt committed:
Rework the wording of the Internals section.

	modified:   README.md
